orbstation/quirks/plurality.dm

```diff
@@ -0,0 +1,316 @@
+		return
+	if(popup_input == "Remove Member")
+		if(!system_member_names.len)
+			to_chat(user, span_warning("ERROR: There are no members registered with [system_name]."))
+			return
+		var/selected_member = tgui_input_list(user, "Select system member to remove:", "Remove Member", system_member_names)
+		if(!selected_member || !(selected_member in system_member_names))
+			return
+		system_member_names -= selected_member
+		to_chat(user, span_notice("Successfully removed member [selected_member] from the chip's associated system."))
+		return
+	if(popup_input == "Edit Member")
+		if(!system_member_names.len)
+			to_chat(user, span_warning("ERROR: There are no members registered with [system_name]."))
+			return
+		var/selected_member = tgui_input_list(user, "Select system member to edit:", "Select Member", system_member_names)
+		if(!selected_member)
+			return
+		var/edited_member_name = tgui_input_text(user, "Input the new name of [selected_member]:", "Edit Member", max_length = 30)
+		if(!edited_member_name)
+			return
+		if(!(selected_member in system_member_names))
+			to_chat(user, span_warning("ERROR: Could not find member [selected_member] in [system_name]."))
+			return
+		system_member_names -= selected_member
+		system_member_names |= edited_member_name
+		to_chat(user, span_notice("Successfully renamed member [selected_member] to [edited_member_name]."))
+		return
+	if(popup_input == "Clear Members")
+		if(!system_member_names.len)
+			to_chat(user, span_warning("ERROR: There are no members registered with [system_name]."))
+			return
+		var/clear_confirmation = tgui_alert(user, "Do you really want to clear all members from the system?", "Clear Members", list("Yes", "No"))
+		if(clear_confirmation == "Yes")
+			system_member_names.Cut()
+			to_chat(user, span_notice("Successfully cleared all members from [system_name]."))
+		return
+	if(popup_input == "Edit System Name")
+		if(system_name_locked)
+			to_chat(user, span_warning("ERROR: This chip cannot have its system name changed without being swiped first by an ID with Head of Personnel access."))
+			return
+		var/new_system_name = tgui_input_text(user, "Input the new system name:", "Edit System Name", max_length = 20)
+		if(!new_system_name)
+			return
+		to_chat(user, span_notice("Successfully changed the chip's associated system name from [system_name] to [new_system_name]."))
+		update_system_name(new_system_name)
+		return
+
+/// Changes the system_name variable and automatically updates the name of the chip to include it.
+/obj/item/plural_system_chip/proc/update_system_name(new_system_name)
+	system_name = new_system_name
+	name = "[initial(name)] {[system_name]}"
+
+/obj/item/plural_system_chip/attackby(obj/item/W, mob/user, params)
+	if(W.GetID())
+		if(allowed(user))
+			locked = !locked
+			system_name_locked = FALSE
+			balloon_alert(user, locked ? "locked" : "unlocked")
+			update_appearance()
+			return
+		else
```

Comment:
Don't need an else following a return.

Reply:
Personally I'd flip some of these to spare on indenting, so:
```
if (!W.getID())
   return
if (!allowed(user)
   balloon_alert("no")
   return
...
```

this is a style preference though